### ISSUE TITLE
Scraping data for shared or replicated profiles

### ISSUE THREAD

**Issue by petekelly** [rrb3942/dialoginsight#1]
Hello, thanks for a great little project - @recourse set me upon it and I think it is going to be very useful!

However some of the dialog profile names that I used are suffixed with /b and /s naming flags as per the documentation for marking binary replicated and shared dialog profiles here https://opensips.org/html/docs/modules/3.2.x/dialog.html

I notice that the scraper does indeed request these profiles and a response is received with the correct profile flag set (replicated:yes in this case) but the data does not end up in prometheus.

Check the request (from dialoginfo to opensips) for `egress/b` below

```
T 2023/09/12 12:29:48.258981 127.0.0.1:39022 -> 127.0.0.1:8888 [AP] #314
{"jsonrpc": "2.0", "id": "12360", "method": "profile_get_size", "params": ["egress/b"]}

T 2023/09/12 12:29:48.259047 127.0.0.1:8888 -> 127.0.0.1:39022 [AP] #316
HTTP/1.1 200 OK.
Connection: close.
Content-Length: 125.
Content-Type: application/json.
Date: Tue, 12 Sep 2023 11:29:48 GMT.
.
{"jsonrpc":"2.0","result":{"Profile":{"name":"egress","value":null,"count":0,"shared":"no","replicated":"yes"}},"id":"12360"}
```

Is this something I can solve with configuration somehow? If not, perhaps these could be exported with a naming convention of <profilename>-b or <profilename>-s into prometheus?

Thanks

Pete

**Comment by rrb3942** [comment]
Hey Pete,

My initial suspicion is that the '/' isn't a valid character for a prometheus label and so the profile is being dropped. I didn't expect OpenSIPs to include the tags in the profile names (which it doesn't in the response to the profile_get_size, so it seems a bit inconsistent)

Could you provide the output of the [list_all_profiles](https://opensips.org/docs/modules/3.2.x/dialog.html#mi_list_all_profiles) MI command? This is how dialoginsight learns all the profile names used for export.

Could you also confirm if doing a 'profile_get_size' against just the profile name without tags ("egress" instead of "egress/b") returns the expected values? I'm thinking that just stripping the tags off the name might be the best approach.

**Comment by petekelly** [comment]
```
root@sv-sbc1:/usr/local/src/dialoginsight# opensips-cli -x mi list_all_profiles
{
    "Profiles": [
        {
            "name": "ingress",
            "has value": false
        },
        {
            "name": "egress",
            "has value": false
        },
        {
            "name": "ccchecks",
            "has value": true
        },
        {
            "name": "ccchecksrepl",
            "has value": true
        }
    ]
}
```

`ccchecksrepl`, `ingress` and `egress` are actually defined within the opensips cfg as /b  but as you see from the above they don't show this tag in the output from `list_all_profiles`

```
modparam("dialog", "profiles_no_value", "ingress/b; egress/b")
modparam("dialog", "profiles_with_value", "ccchecks; ccchecksrepl/b")
```

Calls to `profile_get_size` actually require the `/b` flag in the name:

```
root@sv-sbc1:/usr/local/src/dialoginsight# opensips-cli -x mi profile_get_size ccchecksrepl/b
{
    "Profile": {
        "name": "ccchecksrepl",
        "value": null,
        "count": 2,
        "shared": "no",
        "replicated": "yes"
    }
}
```

```
opensips-cli -x mi profile_get_size ccchecksrepl
ERROR: command 'profile_get_size' returned: 404: Profile not found
```

**Comment by rrb3942** [comment]
Hmm, that is interesting. Where is dialoginsight getting the '/b' from? Do you have that dialog listed in the `export_profiles` option?

**Comment by petekelly** [comment]
I manually specified it in the configuration because it wasn't coming out as part of `export_all`

```
        "insight_label": "insight",
        "export_profiles": ["ccchecks", "ccchecks/b", "ingress/b", "egress/b"],
        "export_all": false,
        "timeout": "3s",
        "idle_remove": "1m",
        "enable_profiling": false
```

**Comment by rrb3942** [comment]
So there are two issues.

1. You can't fetch shared dialogs using the name returned by `list_all_profiles`. This looks like a current limitation of OpenSIPs.
2. When you manually specify shared profiles in `export_profiles` they aren't exported anyways, because we filter it out because of the `/` character.

My initial thought is to use the `name` returned by `profile_get_size` which would at least make #2 work. The remaining question is if `shared` and `replicated` should be included as labels.

**Comment by rrb3942** [comment]
It has not been tested at all yet, but I stared work for this under [shared_profiles](https://github.com/rrb3942/dialoginsight/tree/shared_profiles). Using the 'replication_hints' it should allow remapping the reported profile name to a tagged name. 

Feel free to give that branch a try.

**Comment by rrb3942** [comment]
Thanks for your help in identifying and testing this.

This should now be fixed via the [0.10](https://github.com/rrb3942/dialoginsight/releases/tag/0.10) release. Feel free to reopen if you find that is not the case.
